src/bot/logic/common/eventBus.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Very lightweight in-process event bus for bot logic.
+ */
+
+// --- Event definitions ---------------------------------------------------
+
+export type NavalEvent = {
+    type: "yardFailed";     // a Naval Yard finished production but could not be placed
+    player: string;          // owning player name
+};
+
+export type BotEvent = NavalEvent; // | OtherFutureEvents
+
+// -------------------------------------------------------------------------
+
+type Listener = (event: BotEvent) => void;
+
+const listeners: Set<Listener> = new Set();
```

Comment:
Is this intentionally static? It will be shared across all bot instances.

Reply:
Global state is a potential source for desyncs. Please keep in mind this will be persisted across games and breaks the deterministic nature of the game simulation. Is it really needed to share state between bots? If you must, at least scope it by game ID or make sure it is cleared somehow when the game finishes.